package protocol;

message OpResult
{
	/// 定义缓存服务器和GS 之间的错误消息协议
	enum status_t {
		STATE_NORMAL           = 0;  /// 正常
		STATE_ENTITY_NOT_FOUND = 1;  /// 实体没有找到
		STATE_ENTITY_EXISTS    = 2;  /// 实体已经存在  
		STATE_ENTITY_LOCKED    = 5;  /// 实体被锁定
		
		STATE_PROTOCOL_ERROR  = 104;  /// 协议不正确 
		STATE_INTERNAL_ERROR  = 105;  /// CACHE 服务器内部错误 
	}
	
	required status_t status    = 1;  /// 代表查询状态 
	optional bytes   message   = 2;   /// 代表查询结果，如果存在结果的话

}

message db_data_request
{
	required uint32 transaction_index	= 1;	// 该请求在事物中的索引值
	required uint32 transaction_num		= 2;	// 该请求相关的事物数据总量
	required uint32 transaction_id		= 3;	// 事务唯一编号
	optional uint32 line_id				= 4;	// 数据所在线ID
	optional bytes  data_stream			= 5;	// 实体数据
	optional uint64 owner_guid			= 6;	// 实体对应的玩家guid
}
